iml-manager-cli/src/server.rs

```diff
@@ -419,6 +445,86 @@ pub async fn server_cli(command: ServerCommand) -> Result<(), ImlManagerCliError
 
             wait_for_cmds(cmds).await?;
         }
+        ServerCommand::Remove(config) => {
+            let term = Term::stdout();
+
+            let remove_hosts = hostlist_parser::parse(&config.hosts)?;
+
+            tracing::debug!("Parsed hosts {:?}", remove_hosts);
+
+            let api_hosts = get_hosts().await?;
+
+            let unknown_hosts = filter_unknown_hosts(&remove_hosts, &api_hosts.objects);
+
+            for unknown_host in unknown_hosts {
+                display_cancelled(format!(
+                    "Host {} unknown to IML and will not be removed.",
+                    unknown_host.fqdn
+                ));
+            }
+
+            let objects = remove_hosts
+                .iter()
+                .map(|address| TestHostConfig {
+                    address,
+                    auth_type: "existing_keys_choice".into(),
+                })
+                .collect();
+
+            let Objects { objects } = post("test_host", Objects { objects })
+                .await?
+                .error_for_status()?
+                .json()
+                .await
+                .map_err(iml_manager_client::ImlManagerClientError::Reqwest)?;
+
+            let cmds = objects
+                .into_iter()
+                .map(|CmdWrapper { command }| command)
+                .collect();
+
+            term.write_line(&format!("{} preflight checks...", style("Running").green()))?;
+
+            let _cmds = wait_for_cmds(cmds).await?;
+
+            let objects = remove_hosts
```

Comment:
It looks like all the hosts, even the unknown ones, will be queried. Will the request be successful?

Reply:
Done.